lib/action_subscriber/threadpool.rb

```diff
@@ -4,17 +4,29 @@ class Threadpool
     # Class Methods
     #
     def self.busy?
-      (pool.pool_size == pool.busy_size)
+      pools.any? do |_pool_name, pool|
+        pool.pool_size == pool.busy_size
+      end
     end
 
-    def self.perform_async(*args)
-      self.pool.async.perform(*args)
+    def self.new_pool(name, pool_size = nil)
+      fail ArgumentError, "#{name} already exists as a threadpool" if pools.key?(name)
+      pool_size ||= ::ActionSubscriber.config.threadpool_size
+      pools[name] = ::Lifeguard::InfiniteThreadpool.new(
+        :pool_size => pool_size
+      )
     end
 
-    def self.pool
-      @pool ||= ::Lifeguard::InfiniteThreadpool.new(
-        :pool_size => ::ActionSubscriber.config.threadpool_size
-      )
+    def self.pool(which_pool = :default)
+      pools[which_pool]
+    end
+
+    def self.pools
+      @pools ||= {
+        :default => ::Lifeguard::InfiniteThreadpool.new(
```

Comment:
It feels like maybe this should also be using `self.new_pool` to create the pool so there's not to paths for the same logic. Maybe something like:

```ruby
def self.new_pool
  # same as above
end

def self.pools; @pools ||= {}; end

self.new_pool(:default) unless self.pools.key?(:default)
```

I'd also consider renaming `new_pool` to `add_pool` or just `add`.